src/wp-includes/class-wp-theme.php

```diff
@@ -1860,6 +1860,16 @@ public function get_block_patterns() {
 
 		$files = (array) self::scandir( $dirpath, 'php', -1 );
 
+		/**
+		 * Filters list of block pattern files for a theme.
+		 *
+		 * @since 6.8
+		 *
+		 * @param array  $files   Array of theme files found within `patterns` directory.
+		 * @param string $dirpath Path of theme `patterns` directory being scanned.
+		 */
+		$files = (array) apply_filters( 'get_block_patterns_files', $files, $dirpath );
```

Comment:
Let's rename this to be a bit more consistent with other filters in this file:
```suggestion
		$files = (array) apply_filters( 'theme_block_pattern_files', $files, $dirpath );
```